src/main/java/ch/njol/skript/bukkitutil/BukkitUnsafe.java

```diff
@@ -56,144 +56,31 @@ public class BukkitUnsafe {
-	
-	/**
-	 * If we have material map for this version, using it is preferred.
-	 * Otherwise, it can be used as fallback.
-	 */
-	private static boolean preferMaterialMap = true;
-	
-	/**
-	 * We only spit one exception (unless debugging) from UnsafeValues. Some
-	 * users might not care, and find 1.12 material mappings accurate enough.
-	 */
-	private static boolean unsafeValuesErrored;
-	
+
 	/**
 	 * Maps pre 1.12 ids to materials for variable conversions.
 	 */
 	@Nullable
 	private static Map<Integer,Material> idMappings;
-	
-	public static void initialize() {
-		if (!newMaterials) {
-			MethodHandle mh;
-			try {
-				mh = MethodHandles.lookup().findVirtual(UnsafeValues.class,
-						"getMaterialFromInternalName", MethodType.methodType(Material.class, String.class));
-			} catch (NoSuchMethodException | IllegalAccessException e) {
-				mh = null;
-			}
-			unsafeFromInternalNameMethod = mh;
-			
-			try {
-				Version version = Skript.getMinecraftVersion();
-				boolean mapExists = loadMaterialMap("materials/" + version.getMajor() + "." +  version.getMinor() + ".json");
-				if (!mapExists) {
-					loadMaterialMap("materials/1.9.json"); // 1.9 is oldest we have mappings for
-					preferMaterialMap = false;
-					Skript.warning("Material mappings for " + version + " are not available.");
-					Skript.warning("Depending on your server software, some aliases may not work.");
-				}
-			} catch (IOException e) {
-				Skript.exception(e, "Failed to load material mappings. Aliases may not work properly.");
-			}
-		}
-	}
-	
+
 	@Nullable
 	public static Material getMaterialFromMinecraftId(String id) {
-		if (newMaterials) {
-			// On 1.13, Vanilla and Spigot names are same
-			if (id.length() > 9)
-				return Material.matchMaterial(id.substring(10)); // Strip 'minecraft:' out
-			else // Malformed material name
-				return null;
-		} else {
-			// If we have correct material map, prefer using it
-			if (preferMaterialMap) {
-				if (id.length() > 9) {
-					assert materialMap != null;
-					return materialMap.get(id.substring(10)); // Strip 'minecraft:' out
-				}
-			}
-			
-			// Otherwise, hacks
-			Material type = null;
-			try {
-				assert unsafeFromInternalNameMethod != null;
-				type = (Material) unsafeFromInternalNameMethod.invokeExact(unsafe, id);
-			} catch (Throwable e) {
-				// Only spit out an error once unless debugging
-				if (!unsafeValuesErrored || Skript.debug()) {
-					Skript.exception(e, "UnsafeValues failed to get material from Vanilla id");
-					unsafeValuesErrored = true;
-				}
-			}
-			if (type == null || type == Material.AIR) { // If there is no item form, UnsafeValues won't work
-				// So we're going to rely on 1.12's material mappings
-				assert materialMap != null;
-				return materialMap.get(id);
-			}
-			return type;
-		}
+		// On 1.13, Vanilla and Spigot names are same
+		if (id.length() > 9)
+			return Material.matchMaterial(id.substring(10)); // Strip 'minecraft:' out
```

Comment:
I am concerned about the assumption that the ID will start with `minecraft:` (in case of a future where some don't) - might it be worth substringing from the index of `:` + 1 in case it saves a future headache?

Reply:
This method is called `getMaterialFromMinecraftId`, so I this should only be used for Minecraft IDs. Anyway, this is outside the scope of this PR (i.e. removing legacy code)